src/test/java/com/arpnetworking/tsdcore/sinks/DimensionFilteringSinkTest.java

```diff
@@ -86,6 +86,32 @@ public void testInjectExcludeWithout() {
         Mockito.verify(_target).recordAggregateData(dataIncluded);
     }
 
+    @Test
+    public void testWhistlist() {
+        final DimensionFilteringSink sink = new DimensionFilteringSink.Builder()
+                .setName("testWhistlist")
+                .setSink(_target)
+                .setWhitelistDimensions(ImmutableSet.of("foo", "bar"))
+                .build();
+        final PeriodicData dataExcluded = TestBeanFactory.createPeriodicDataBuilder()
+                .setDimensions(ImmutableMap.of("foo", "fooVal", "baz", "bazVal"))
+                .build();
+        sink.recordAggregateData(dataExcluded);
+        Mockito.verify(_target, Mockito.never()).recordAggregateData(Mockito.any(PeriodicData.class));
+
+        final PeriodicData dataIncluded = TestBeanFactory.createPeriodicDataBuilder()
```

Comment:
Your code does a containsAll, so it fails not because it has abc, but because it is missing bar. My point is that your sink ignores additional dimensions keys. This may go back to your intended purpose for this filter, can you elaborate on that?